pow/src/test/java/tech/pegasys/artemis/pow/DepositRequestManagerTest.java

```diff
@@ -0,0 +1,206 @@
+  }
+
+  @Test
+  void checkTheRangeAskedForIsCorrect() {
+    mockLatestCanonicalBlockNumber(10);
+    depositRequestManager.start();
+    ArgumentCaptor<DefaultBlockParameter> fromBlockArgumentCaptor =
+        ArgumentCaptor.forClass(DefaultBlockParameter.class);
+    ArgumentCaptor<DefaultBlockParameter> toBlockArgumentCaptor =
+        ArgumentCaptor.forClass(DefaultBlockParameter.class);
+    verify(depositContract)
+        .depositEventEventsInRange(
+            fromBlockArgumentCaptor.capture(), toBlockArgumentCaptor.capture());
+
+    assertThat(Numeric.decodeQuantity(fromBlockArgumentCaptor.getValue().getValue())).isEqualTo(0);
+    assertThat(Numeric.decodeQuantity(toBlockArgumentCaptor.getValue().getValue())).isEqualTo(10);
+  }
+
+  @Test
+  void depositsInConsecutiveBlocks() {
+    mockLatestCanonicalBlockNumber(10);
+
+    mockContractEventsInRange(
+        0,
+        10,
+        List.of(
+            mockDepositEventEventResponse(1, "0x1234", 1),
+            mockDepositEventEventResponse(2, "0x1234", 1),
+            mockDepositEventEventResponse(3, "0x2345", 2)));
+
+    mockBlockForEth1Provider("0x1234", 1, 1000);
+    mockBlockForEth1Provider("0x2345", 2, 1014);
+
+    depositRequestManager.start();
+
+    ArgumentCaptor<DepositsFromBlockEvent> eventArgumentCaptor =
+        ArgumentCaptor.forClass(DepositsFromBlockEvent.class);
+    verify(depositEventChannel, times(2)).notifyDepositsFromBlock(eventArgumentCaptor.capture());
+    assertThat(eventArgumentCaptor.getAllValues()).hasSize(2);
+    assertThat(
+            eventArgumentCaptor.getAllValues().stream()
+                .map(DepositsFromBlockEvent::getBlockNumber)
+                .map(UnsignedLong::longValue)
+                .collect(Collectors.toList()))
+        .containsExactly(1L, 2L);
+  }
+
+  @Test
+  void noDepositsInSomeBlocksInRange() {
+    mockLatestCanonicalBlockNumber(10);
+
+    mockContractEventsInRange(
+        0,
+        10,
+        List.of(
+            mockDepositEventEventResponse(1, "0x1234", 1),
+            mockDepositEventEventResponse(2, "0x1234", 1),
+            mockDepositEventEventResponse(3, "0x2345", 2),
+            mockDepositEventEventResponse(4, "0x4567", 4)));
+
+    mockBlockForEth1Provider("0x1234", 1, 1000);
+    mockBlockForEth1Provider("0x2345", 2, 1014);
+    mockBlockForEth1Provider("0x4567", 4, 1042);
+
+    depositRequestManager.start();
+
+    ArgumentCaptor<DepositsFromBlockEvent> eventArgumentCaptor =
+        ArgumentCaptor.forClass(DepositsFromBlockEvent.class);
+    verify(depositEventChannel, times(3)).notifyDepositsFromBlock(eventArgumentCaptor.capture());
+    assertThat(eventArgumentCaptor.getAllValues()).hasSize(3);
+    assertThat(
+            eventArgumentCaptor.getAllValues().stream()
+                .map(DepositsFromBlockEvent::getBlockNumber)
+                .map(UnsignedLong::longValue)
+                .collect(Collectors.toList()))
+        .containsExactly(1L, 2L, 4L);
```

Comment:
It's better to use an argument matcher to do this kind of verification of method calls rather than an ArgumentCaptor.  Captors are more intended for things like listeners where you need to call methods on them to progress the test.  So:

```
    verify(depositEventChannel).notifyDepositsFromBlock(argThat(isEventWithBlockNumber(1)));
    verify(depositEventChannel).notifyDepositsFromBlock(argThat(isEventWithBlockNumber(2)));
    verify(depositEventChannel).notifyDepositsFromBlock(argThat(isEventWithBlockNumber(4)));
    verifyNoMoreInteractions(depositEventChannel);
```
where isEventWithBlockNumber is:
```
  private ArgumentMatcher<DepositsFromBlockEvent> isEventWithBlockNumber(
      final long expectedBlockNumber) {
    return argument -> argument.getBlockNumber().longValue() == expectedBlockNumber;
  }
```

You may want to extend the matcher to also assert at least the expected number of deposits in each event.

Reply:
Neat.

Reply:
done.